flow/src/main/java/com/sequenceiq/flow/core/Flow2Handler.java

```diff
@@ -562,7 +561,16 @@ private void logFlowId(String flowId) {
         LOGGER.debug("Flow has been created with id: '{}' and the related request id: '{}'.", flowId, requestId);
     }
 
-    public FlowLog getFirstStateLogfromLatestFlow(Long resourceId) {
+    public FlowLog getFirstRetryableStateLogfromLatestFlowOrThrow(Long resourceId) {
```

Comment:
I know I suggested the "throw" part in method name, but this looks little bit odd, maybe throw par should not be part of method name, only this would be enough: `getFirstRetryableStateLogFromLatestFlow`. what do you think?

Reply:
I think either works, it should be understand from looking at the method and that it's not returning the optional. also "FlowOrThrow" is a mouthful